client/src/main/java/io/pravega/client/ClientConfig.java

```diff
@@ -122,12 +122,22 @@ private void extractCredentials() {
         @VisibleForTesting
         ClientConfigBuilder extractCredentials(Properties properties, Map<String, String> env) {
             if (credentials != null) {
+                log.debug("Client credentials were specified by the caller.");
                 return this;
             }
-
-            credentials = extractCredentialsFromProperties(properties);
-            if (credentials == null) {
+            if (properties != null) {
+                credentials = extractCredentialsFromProperties(properties);
+                if (credentials != null) {
+                    log.debug("Client credentials were extracted from system properties.");
+                    return this;
+                }
+            }
+            if (env != null) {
                 credentials = extractCredentialsFromEnv(env);
+                if (credentials != null) {
+                    log.debug("Client credentials were extracted from environment variables");
```

Comment:
Since this is for debugging purposes, do you want to list as part of the log message the variables that have been extracted? Same comment for the other log messages in this file.

Reply:
Since credentials are considered sensitive, they are usually not logged. It might be helpful to know whether they are present, and therefore I added the current log statements, which I think should establish conclusively whether the credentials were present. 

What do you say?

Reply:
Ok.